pallets/phala/src/compute/vault.rs

```diff
@@ -410,8 +410,19 @@ pub mod pallet {
 						)
 						.expect("get nft should not fail: qed.");
 						let property = &property_guard.attr;
+						if !base_pool::is_nondust_balance(property.shares) {
```

Comment:
You are burning the share from a pool (a Stake Pool in this case). Do you ensure the stake pool total share also decrease?

Reply:
Looks like a few `burn_nft()` call in BasePool is also not properly handled?

Reply:
Have you considered to handle the dust problem in `stake_pool_v2::Pallet::<T>::withdraw()`?